### PR TITLE
Bump to Django 1.8.4

### PR DESCRIPTION
**Pull request by kamarkiewicz**
Bump to Django 1.8.4.
No database manipulations - my PR needs an update of README.md about db.
Improved SECRET_KEY handling. Well, I decided to use Django's key generator.
Minor fixes, code refactoring and improvements.

Tested on python-2.7 and python-3.3 cartridges on OpenShift.com.

Based on #47 (thanks to @soltysh and @mfojtik)

### CODE REVIEW

File: wsgi/application [hunk 1]
```diff
@@ -7,14 +7,8 @@ import sys
 os.environ['DJANGO_SETTINGS_MODULE'] = 'myproject.settings'
 ## GETTING-STARTED: make sure the next line points to your django project dir:
 sys.path.append(os.path.join(os.environ['OPENSHIFT_REPO_DIR'], 'wsgi', 'myproject'))
-virtenv = os.environ['APPDIR'] + '/virtenv/'
```
Comment (kamarkiewicz) [inline wsgi/application]: In Python cartridges virtualenv is activated automatically durning build and deploy. (as I observed so far)

File: .openshift/action_hooks/deploy [hunk 2]
```diff
@@ -4,24 +4,17 @@
 # up again.  This script gets executed directly, so it could be python, php,
 # ruby, etc.
 
-source $OPENSHIFT_HOMEDIR/python/virtenv/bin/activate
```
Comment (kamarkiewicz) [inline .openshift/action_hooks/deploy]: In Python cartridges virtualenv is activated automatically durning build and deploy. (as I've observed so far)

**Comment by soltysh** [comment]
I'll review that later today. Thanks @kamarkiewicz!

@rhcarvalho fyi

File: .openshift/action_hooks/deploy (hunk 2 above)
Comment (soltysh) [inline .openshift/action_hooks/deploy]: Correct it's done during [build](https://github.com/openshift/origin-server/blob/master/cartridges/openshift-origin-cartridge-python/usr/versions/shared/bin/control#L177) and [deploy](https://github.com/openshift/origin-server/blob/master/cartridges/openshift-origin-cartridge-python/usr/versions/shared/bin/control#L195) steps.

**Comment by soltysh** [comment]
LGTM

**Pull request merged**